File: backend/application/operator.py
from rest_framework import serializers
from domain.operator import calcular_ggr, calcular_exposure, generar_reporte_csv


class MetricsSerializer(serializers.Serializer):
    total_stakes = serializers.CharField()
    total_payouts = serializers.CharField()
    ggr = serializers.CharField()
    total_bets = serializers.IntegerField()
    active_users = serializers.IntegerField(required=False, default=0)
    active_bets = serializers.IntegerField(required=False, default=0)


class ExposureItemSerializer(serializers.Serializer):
    selection_id = serializers.IntegerField()
    selection_name = serializers.CharField()
    odds = serializers.CharField()
    bets_count = serializers.IntegerField()
    potential_payout = serializers.CharField()

File: frontend/src/services/operator.ts
import api from './api'

export interface OperatorMetrics {
  total_stakes: string
  total_payouts: string
  ggr: string
  total_bets: number
  active_users: number
  active_bets: number
}

export interface ExposureItem {
  selection_id: number
  selection_name: string
  odds: string
  bets_count: number
  potential_payout: string
}

export const operator = {
  getMetrics: async (): Promise<OperatorMetrics> => {
    const response = await api.get('/operator/metrics/')
    return response.data
  },

  getExposure: async (eventId: number): Promise<ExposureItem[]> => {
    const response = await api.get(`/operator/exposure/${eventId}/`)
    return response.data
  },

  getReportURL: (mes: number, anio: number): string => {
    const params = new URLSearchParams({ mes: String(mes), anio: String(anio) })
    return `/api/operator/reporte/?${params.toString()}`
  },
}
